// vue-lsi-util/APIAccesoV2.js
import store from '@/store'
import axios from 'axios'

const AccesosAAPIV2 = {
    async acceder (payload) {
        return new Promise( 
            function (resolve, reject) {
                if (typeof(payload.Cartel)!=='undefined') {
                    store.dispatch('loading/mostrar', payload.Cartel)
                }
                if (typeof(payload.Metodo)=="undefined") {
                    payload.Metodo="GET"
                }
                if (typeof(payload.Cabeceras) === "undefined") {
                    payload.Cabeceras = {};
                }
                if (typeof(payload.Credenciales) === "undefined") {
                    payload.Credenciales = JSON.parse(process.env.VUE_APP_API).CredencialesV2
                }
                const prefijoURL=JSON.parse(process.env.VUE_APP_API).URLV2
        
                axios(
                    {
                        method: payload.Metodo, 
                        url: prefijoURL+payload.Ruta, 
                        headers: payload.Cabeceras, 
                        data: payload.Body,
                        auth: payload.Credenciales
                    }
                )
                .then(datos => {
                    if (datos.data.status.toUpperCase()=="OK") {
                        resolve(datos.data.data)
                    } else {
                        reject(datos.data.errors)
                    }
                })
                .catch(error => {
                    reject(error)
                })
                .finally( () => {
                    if (typeof(payload.Cartel)!=='undefined') {
                        store.dispatch('loading/ocultar')
                    }        
                })
            }
        )
    }
}

export default AccesosAAPIV2
